model/transformers/dffml_model_transformers/ner/ner_model.py

```diff
@@ -168,38 +162,51 @@ class NERModelConfig:
     )
 
     def __post_init__(self):
+        self.tf = importlib.import_module("tensorflow")
         if self.use_fp16:
-            tf.config.optimizer.set_experimental_options(
+            self.tf.config.optimizer.set_experimental_options(
                 {"auto_mixed_precision": True}
             )
         if self.tpu:
-            resolver = tf.distribute.cluster_resolver.TPUClusterResolver(
+            resolver = self.tf.distribute.cluster_resolver.TPUClusterResolver(
                 tpu=self.parent.config.tpu
             )
-            tf.config.experimental_connect_to_cluster(resolver)
-            tf.tpu.experimental.initialize_tpu_system(resolver)
-            self.strategy = tf.distribute.experimental.TPUStrategy(resolver)
+            self.tf.config.experimental_connect_to_cluster(resolver)
+            self.tf.tpu.experimental.initialize_tpu_system(resolver)
+            self.strategy = self.tf.distribute.experimental.TPUStrategy(
+                resolver
+            )
             self.n_device = self.num_tpu_cores
         elif len(self.gpus.split(",")) > 1:
             self.n_device = len(
                 [f"/gpu:{gpu}" for gpu in self.gpus.split(",")]
             )
-            self.strategy = tf.distribute.MirroredStrategy(
+            self.strategy = self.tf.distribute.MirroredStrategy(
                 devices=[f"/gpu:{gpu}" for gpu in self.gpus.split(",")]
             )
         elif self.no_cuda:
             self.n_device = 1
-            self.strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+            self.strategy = self.tf.distribute.OneDeviceStrategy(
+                device="/cpu:0"
+            )
         else:
             self.n_device = len(self.gpus.split(","))
-            self.strategy = tf.distribute.OneDeviceStrategy(
+            self.strategy = self.tf.distribute.OneDeviceStrategy(
                 device="/gpu:" + self.gpus.split(",")[0]
             )
 
 
 class NERModelContext(ModelContext):
     def __init__(self, parent, **kwconfig):
         super().__init__(parent)
+        self.pd = importlib.import_module("pandas")
+        self.np = importlib.import_module("numpy")
+        self.tf = importlib.import_module("tensorflow")
+        self.re = importlib.import_module("re")
+        self.math = importlib.import_module("math")
+        self.glob = importlib.import_module("glob")
+        self.datetime = importlib.import_module("datetime")
+        self.collections = importlib.import_module("collections")
```

Comment:
Same thing, re, math, glob, datetime, and collections, are in the stdlib. Also, we can replace glob with `pathlib(the_path).glob(arguments)`